ratatui-widgets/src/table.rs

```diff
@@ -824,6 +824,48 @@ impl Table<'_> {
         }
     }
 
+    /// Return the space that a Cell should occupy, taking into account its colspans
+    ///
+    /// Returns `None` when there are no more columns for the Cell to occupy; in this case,
+    /// rendering of the rest of the row can safely be skipped.
+    /// Otherwise, returns a pair `Some((x, width))`, representing the start x-coordinate and width
+    /// of the Cell.
+    fn get_widths_for_cell<'a, T>(column_widths_iterator: &mut T, cell: &Cell) -> Option<(u16, u16)>
+    where
+        T: Iterator<Item = &'a (u16, u16)>,
+    {
+        let mut ret = None;
```

Comment:
I'd prefer if we define an internal type and use that instead of using a tuple here. e.g. `CellSize` that holds horizontal position (x) and total width of the `Cell`.

This would also make the code more readable (e.g. we could use a more definitive name instead of just saying `ret`)